### PR TITLE
Query all DNS seeders if missing many connections 

### PR DESCRIPTION
**Pull request by michaelsutton**

### CODE REVIEW

File: components/connectionmanager/src/lib.rs [hunk 1]
```diff
@@ -227,12 +227,14 @@ impl ConnectionManager {
...
-                cmgr.dns_seed(missing_connections); //TODO: Consider putting a number higher than `missing_connections`.
-            })
-            .await;
+            if missing_connections > self.outbound_target / 2 {
+                // If we are missing more than half of our target, query all in parallel.
+                // This will always be the case on new node start-up and is the most resilient strategy in such a case.
+                self.dns_seed_many(self.dns_seeders.len()).await;
+            } else {
+                // Try to obtain at least twice the number of missing connections
+                self.dns_seed_sequential(2 * missing_connections).await;
```
Comment (biryukovmaxim) [inline components/connectionmanager/src/lib.rs:236]: naming is not clear, both variants adds many connections. 
my suggestion is `dns_seed_many` -> `dns_seed_in_parallel`

File: components/connectionmanager/src/lib.rs [hunk 2]
```diff
@@ -279,6 +271,42 @@ impl ConnectionManager {
...
+    }
+
+    /// Query a single DNS seeder and add the obtained addresses to the address manager.
+    ///
+    /// DNS lookup is a blocking i/o operation so this function is assumed to be called
+    /// from a blocking execution context.
+    fn dns_seed_single(self: &Arc<Self>, seeder: &str) -> usize {
+        info!("Querying DNS seeder {}", seeder);
+        // Since the DNS lookup protocol doesn't come with a port, we must assume that the default port is used.
+        let addrs = match (seeder, self.default_port).to_socket_addrs() {
```
Comment (biryukovmaxim) [inline components/connectionmanager/src/lib.rs:292]: what about async call her with: 
https://docs.rs/tokio/latest/tokio/net/fn.lookup_host.html https://docs.rs/tokio/latest/tokio/net/trait.ToSocketAddrs.html


maybe unrelated change, does it worth TODO, wdyt?

Comment (michaelsutton) [reply components/connectionmanager/src/lib.rs:292]: Cool. I think we were looking for something like this. I'll give it a try now

Comment (biryukovmaxim) [reply components/connectionmanager/src/lib.rs:292]: I guess it doesn't work the way I thought about

Comment (michaelsutton) [reply components/connectionmanager/src/lib.rs:292]: You tried?

File: components/connectionmanager/src/lib.rs (hunk 1 above)
Comment (michaelsutton) [reply components/connectionmanager/src/lib.rs:236]: They are also different in the way they decide to break (min addresses or num seeders), I think parallel vs. seq is misleading. Any other better suggestion?

Comment (biryukovmaxim) [reply components/connectionmanager/src/lib.rs:236]: `_by_seeders[_count]` and `_by_connections[_count]`

File: components/connectionmanager/src/lib.rs (hunk 2 above)
Comment (biryukovmaxim) [reply components/connectionmanager/src/lib.rs:292]: In rust playground. I only got errors.

That's what we can use:
https://docs.rs/trust-dns-resolver/0.23.2/trust_dns_resolver/struct.Resolver.html#method.lookup_ip

Comment (michaelsutton) [reply components/connectionmanager/src/lib.rs:292]: That's for another day

File: components/connectionmanager/src/lib.rs (hunk 1 above)
Comment (michaelsutton) [reply components/connectionmanager/src/lib.rs]: Done
